### PR TITLE
Fix issue where small tooltips could leave their trigger

### PR DESCRIPTION
**Pull request by snowystinger**
Comment all the code for positioning so others can follow it
fix tests
add stories so we can compare positions of tooltips and popovers

Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/overlays/src/calculatePosition.ts [hunk 1]
```diff
@@ -191,18 +191,29 @@ function computePosition(
...
+    // at this point the overlay center should match the button center
     position[crossAxis] += (childOffset[crossSize] - overlaySize[crossSize]) / 2;
   } else if (crossPlacement !== crossAxis) {
+    //  + (button size) - (overlay size)
+    // at this point the overlay bottom should match the button bottom
     position[crossAxis] += (childOffset[crossSize] - overlaySize[crossSize]);
-  }
+  }/* else {
+    the overlay top should match the button top
+  } */
```
Comment (ktabors) [inline packages/@react-aria/overlays/src/calculatePosition.ts:206]: I assume this is just explaining what happens when the conditionals don't match.

File: packages/@react-spectrum/dialog/stories/DialogTrigger.stories.tsx [hunk 2]
```diff
@@ -305,6 +306,31 @@ storiesOf('DialogTrigger', module)
...
     'alert dialog',
     () => renderAlert({})
+  )
+  .add(
+    'crossoffset examples',
+    () => (
+      <Flex gap="size-200" alignSelf="center">
+        <Flex gap="size-200" direction="column" alignItems="start">
+          <span>Left Top</span>
+          {renderPopover({type: 'popover', placement: 'left top', crossOffset: -50}, false)}
```
Comment (ktabors) [inline packages/@react-spectrum/dialog/stories/DialogTrigger.stories.tsx:316]: Can you add the crossOffset to the trigger label for all of these?

File: packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx [hunk 3]
```diff
@@ -100,6 +100,94 @@ storiesOf('TooltipTrigger', module)
         <Button variant="secondary">No Tooltip</Button>
       </Flex>
     )
+  )
+  .add(
+    'crossoffset examples',
```
Comment (ktabors) [inline packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx:105]: I number of these seem positioned wrong in Chrome OSX. I attached to images, but more do these two cases for me.
image
image

**Review by ktabors — CHANGES_REQUESTED** [review]
Love all the comments, I attached some images of a few tooltips that look weird to me.

File: packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx (hunk 3 above)
Comment (snowystinger) [reply packages/@react-spectrum/tooltip/stories/TooltipTrigger.stories.tsx:105]: yeah, i made massive offsets to illustrate that it wouldn't leave, but arrow size is unknown by the overlay, so i think devon's thought was this is fine, we'll trust people not to go bananas

**Pull request merged**